### PR TITLE
chore(CPlusPlus): add avl tree

### PR DESCRIPTION
**Pull request by Ashborn-SM**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [x] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [ ] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**
Add avl tree

**Other information:**

### CODE REVIEW

File: algorithms/CPlusPlus/Trees/avl.cpp [hunk 1]
```diff
@@ -0,0 +1,187 @@
+#include <iostream>
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Trees/avl.cpp:1]: add the description about the program

File: algorithms/CPlusPlus/Trees/avl.cpp [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+	TreeNode* left;
+	TreeNode* right;
+	int data, height;
+	TreeNode(const int& data): data(data), left(nullptr), right(nullptr), height(0){}
+};
+ 
+int height(TreeNode* node){
+	/**
+	 * Return the height of the node
+	 */
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Trees/avl.cpp:13]: use single line comment where as possible

File: algorithms/CPlusPlus/Trees/avl.cpp [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	rightrotation(node);
+}
+
+void insert(TreeNode*& root, const int& data){
+	/**
+	 * Create and insert the node in the appropriate place of the tree
+	 * Update the height of every node in the tree
+	 * Check the balance factor of the node and do the appropriate rotation
+	 * to make the tree balanced
+	 */
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Trees/avl.cpp:101]:
```suggestion
	/*
	 * Create and insert the node in the appropriate place of the tree
	 * Update the height of every node in the tree
	 * Check the balance factor of the node and do the appropriate rotation
	 * To make the tree balanced
	 */
```

File: algorithms/CPlusPlus/Trees/avl.cpp [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+	  / \   \
+	 5   7	 10
+
+	Output: 4 2 1 3 8 6 5 7 9 10
+	Here the balance factor of the root node is 1
+
+	*/
+
+	print(root);
+}
```
Comment (UG-SEP) [inline algorithms/CPlusPlus/Trees/avl.cpp:187]: add time complexity of the program at the EOF

**Review by UG-SEP — CHANGES_REQUESTED** [review]
Please kindly do the changes 
Feel free to ask doubt related to this PR

**Review by UG-SEP — APPROVED** [review]
thanks for contributing @Ashborn-SM

**Review by ming-tsai — APPROVED** [review]

**Pull request merged**